### DISCUSSION TITLE
How to I get single <esc> to do what it does in vanilla vim? Seems that currently <esc>+key is bound to <A-key>.

### DISCUSSION THREAD

**Discussion by DTchebotarev** [LazyVim/LazyVim#163] [Q&A] [answered]
I just installed LazyVim and I'm trying to make some configurations, but having trouble figuring out where to do it.

In insert mode, hitting `<esc>j` moves a line down. I'm guessing there's a plugin somewhere that converts the `<esc>j` into `<A-j>` which is mapped to move line down. How can I figure out which plugin this is? `verbose imap <esc>j` or `verbose imap <esc>` aren't showing anything.

**Comment by folke** [comment]
This is super weird indeed. I can reproduce it by quickly typing `<esc>j` or `<esc>k` in insert mode.

The mapppings that seem to cause this are indeed the ones below:

```lua
map("i", "<A-j>", "<Esc>:m .+1<cr>==gi", { desc = "Move down" })
map("i", "<A-k>", "<Esc>:m .-2<cr>==gi", { desc = "Move up" })
```

I have no idea what could cause this.

**Comment by folke** [comment, marked as answer]
Seems to be a Neovim bug/feature. Can easily be reproduced without LazyVim.

Create `repro.lua`

```lua
local map = vim.keymap.set
map("i", "<A-j>", "<Esc>:m .+1<cr>==gi", { desc = "Move down" })
map("i", "<A-k>", "<Esc>:m .-2<cr>==gi", { desc = "Move up" })
```

Then run `nvim -u ./repro.lua`

**Reply by DTchebotarev** [reply]
So, confirmed this is not a lazyvim (or even neovim) thing. 

This is a known tmux "feature": https://github.com/neovim/neovim/wiki/FAQ#esc-in-tmux-or-gnu-screen-is-delayed

However even the suggested fix doesn't work for me - even setting esc-time to 0 in ~/.tmux.conf I can still produce the `<esc>j` => `<A-j>` behavior.

Running lazyvim outside of tmux, I can't reproduce the behavior at all.

Thanks for looking into it.

**Reply by folke** [reply]
I can reproduce the issue in wezterm without tmux

**Reply by DTchebotarev** [reply]
I'm using kitty terminal as my terminal which doesn't have this issue. Terminal.app does have it though. I guess it's your terminal app and/or tmux that can cause it, by the time the keystrokes get to neovim it's already `<A-key>` instead of `<esc>key` and neovim can't do much about that.

**Reply by loiccoyle** [reply]
I use `alacritty` and I can't reproduce.

**Reply by zoopp** [reply]
Odd, I'm using alacritty (`alacritty 0.11.0 (8dbaa0bb)`) and I CAN reproduce it. How would I go about disabling this behavior?

**Reply by folke** [reply]
It's clearly an issue with the terminal or multiplexer you're using, so check its docs? Google the issue?

**Reply by graciasc** [reply]
It seems like it's a tmux issue.

This reddit post should give you some more context / solution.


https://reddit.com/r/lunarvim/s/Z19qBZbOJ0

**Reply by zhongjis** [reply]
I tried @graciasc 's reddit link and added first part of `u/Agreeable-Vanilla-76` 's solution fixed my issue.
```
set -sg escape-time 0 
set -g status-interval 0
```
